### PR TITLE
[CDE-837] - FilterComponent - html injection

### PR DESCRIPTION
**Pull request by YuryBY**
@velasques, @diogofscmariano, @pamval, could you please take a look? - the fix is without tests since the issue is the blocker for codefreeze. Tests will be provided later.

### CODE REVIEW

**Comment by YuryBY** [comment]
@krivera-pentaho, @jwarlander, adding to the thread

File: core-js/src/main/javascript/cdf/components/filter/HtmlUtils.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+define(function(){
+  return {
+    sanitizeHtml: function(html){	  
+	  html = html.replace(/<script\b[^>]*>/g, "&lt;script&gt;").replace(/<\/script>/g, "&lt;/script&gt;");
+	  html = html.replace(/<iframe\b[^>]*>/g, "&lt;script&gt;").replace(/<\/iframe>/g, "&lt;/script&gt;");
```
Comment (carlosrusso) [inline core-js/src/main/javascript/cdf/components/filter/HtmlUtils.js]: Should we also test for </html> or </body> ?

File: core-js/src/main/javascript/cdf/components/filter/views/Abstract.js [hunk 2]
```diff
@@ -126,7 +128,9 @@ define([
       return this;
     },
     renderSkeleton: function (viewModel) {
-      this.$el.html(Mustache.render(this.template.skeleton, viewModel));
+	  var rHtml = Mustache.render(this.template.skeleton, viewModel)
```
Comment (krivera-pentaho) [inline core-js/src/main/javascript/cdf/components/filter/views/Abstract.js]: Missing ;

File: core-js/src/main/javascript/cdf/components/filter/HtmlUtils.js (hunk 1 above)
Comment (YuryBY) [inline core-js/src/main/javascript/cdf/components/filter/HtmlUtils.js]: @carlosrusso I've done. But I believe <html> and <body> tags do not suppose injection.

**Comment by YuryBY** [comment]
@krivera-pentaho, the PR is updated

**Comment by YuryBY** [comment]
@krivera-pentaho, @jwarlander, @carlosrusso, @diogofscmariano, PR is updated, tests are added

**Comment by YuryBY** [comment]
@pamval, the PR is updated

**Pull request merged**